### PR TITLE
feat: component-testing mode, initial commit

### PR DESCRIPTION
**Pull request by elevatebart**
### ⚠️ Pre-pre alpha feature ⚠️

In this new mode, cypress launches a webpack dev server to run your components

If you run `cypress open --component-testing`, this is what the interface looks like right now:

Screen Shot 2020-12-03 at 3 24 24 PM

### CODE REVIEW

File: packages/runner-ct/src/specs/specs-list.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+@observer
+class SpecsList extends Component {
+  render () {
+    const specGroups = specsStore.specs.length ? makeSpecHierarchy(specsStore.specs) : {}
+
+    return (
+      <div className="specs-list">
+        <header>Select tests to run...</header>
+        <StateContext.Provider value={{
```
Comment (dmtrKovalenko) [inline packages/runner-ct/src/specs/specs-list.jsx:15]: Let's not mix up context providers and mobx state. Unless we decided to completely rewrite test runner state lets use mobx as single source of trust

File: packages/runner-ct/src/specs/specs-list.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      if (i < pathArray.length - 1) {
+        currentGroup[pathPart] = tmpGroup
+        currentGroup = tmpGroup
+      } else {
+        currentGroup[pathPart] = spec
+      }
+    })
+  })
+
+  return groups
```
Comment (dmtrKovalenko) [inline packages/runner-ct/src/specs/specs-list.jsx:76]: Why not to use array.reduce here?

**Review by dmtrKovalenko — CHANGES_REQUESTED** [review]

File: packages/runner-ct/src/specs/spec-group.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            key={spec}
+            path={`${parentPath}/${groupKey}`}
+            spec={group[spec]}/>
+          : <SpecGroup
+            key={spec}
+            groupKey={spec}
+            group={group[spec]}
+            parentPath={`${parentPath}/${groupKey}`}/>)}
+    </ul>
+  </li>)
```
Comment (dmtrKovalenko) [inline packages/runner-ct/src/specs/spec-group.jsx:28]: We need conventions about code formatatting. I am not sure why cypress is not using prettier, but we need at least eslint for jsx formatting

Comment (elevatebart) [reply packages/runner-ct/src/specs/spec-group.jsx:28]: +1

File: packages/runner-ct/src/specs/specs-list.jsx (hunk 2 above)
Comment (elevatebart) [reply packages/runner-ct/src/specs/specs-list.jsx:76]: Lazyness...

Comment (elevatebart) [reply packages/runner-ct/src/specs/specs-list.jsx:76]: I will update and use a reducer.

File: packages/runner-ct/src/specs/spec-group.jsx (hunk 3 above)
Comment (elevatebart) [reply packages/runner-ct/src/specs/spec-group.jsx:28]: Seems like @brian-mann added a `.prettierignore` file 2 years ago...
This file contains `**/*`. 
If we want to adopt prettier, we might need to talk with him.

File: packages/runner-ct/src/specs/specs-list.jsx (hunk 2 above)
Comment (elevatebart) [reply packages/runner-ct/src/specs/specs-list.jsx]: updated

File: packages/runner-ct/src/specs/specs-list.jsx (hunk 1 above)
Comment (elevatebart) [reply packages/runner-ct/src/specs/specs-list.jsx]: It makes the code look more complicated that it should but I agree, let's use mobx as the single source of truth.

**Comment by dmtrKovalenko** [comment]
Nice, thanks for getting it to the review-able state 🐣

**Comment by elevatebart** [comment]
PR abandoned in favor of #14479

**Pull request closed without merging**